src/mdns/mdns_avahi.cpp

```diff
@@ -378,10 +367,7 @@ void PublisherAvahi::HandleGroupState(AvahiEntryGroup *aGroup, AvahiEntryGroupSt
 
 void PublisherAvahi::HandleGroupState(AvahiEntryGroup *aGroup, AvahiEntryGroupState aState)
```

Comment:
I think the service is not actually multicasted on the link immediately after calling `avahi_entry_group_commit`
when publishing a service (mDNS needs to do probing before announcing new RRs).

The Advertising Proxy waits for the publisher to call the `mServiceHandler` to notify the completion of the mDNS registration:
https://github.com/openthread/ot-br-posix/blob/a1e0c7344e137bda1143bf72a252e36f60d392e6/src/mdns/mdns.hpp#L335-L339

I think the proper place to call the `mServiceHandler` is at here. Thoughts?

Reply:
Good catch, I forgot about the handlers. Thanks!